airbyte-integrations/connectors/destination-pinecone/destination_pinecone/config.py

```diff
@@ -0,0 +1,122 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+import re
+from typing import List, Literal, Optional, Union
+
+from jsonschema import RefResolver
+from pydantic import BaseModel, Field
+
+
+class ProcessingConfigModel(BaseModel):
+    chunk_size: int = Field(
+        ...,
+        title="Chunk size",
+        maximum=8191,
+        description="Size of chunks in tokens to store in vector store (make sure it is not too big for the context if your LLM)",
+    )
+    chunk_overlap: int = Field(
+        title="Chunk overlap",
+        description="Size of overlap between chunks in tokens to store in vector store to better capture relevant context",
+        default=0,
+    )
+    text_fields: Optional[List[str]] = Field(
+        ...,
+        title="Text fields to embed",
+        description="List of fields in the record that should be used to calculate the embedding. All other fields are passed along as meta fields. The field list is applied to all streams in the same way and non-existing fields are ignored. If none are defined, all fields are considered text fields. When specifying text fields, you can access nested fields in the record by using dot notation, e.g. `user.name` will access the `name` field in the `user` object. It's also possible to use wildcards to access all fields in an object, e.g. `users.*.name` will access all `names` fields in all entries of the `users` array.",
+        always_show=True,
+        examples=["text", "user.name", "users.*.name"],
+    )
+    meta_data_fields: Optional[List[str]] = Field(
```

Comment:
Nit: meta_data_fields here, but metadata_fields later in the class, could we unfiy?